src/construct.ts

```diff
@@ -77,7 +77,12 @@ export class Node {
    * Components are separated by '/'.
    */
   public get path(): string {
-    const components = this.scopes.filter(c => c.node.id).map(c => c.node.id);
```

Comment:
Calling `filter` generates one array, and calling `map` after creates another array. It should be slightly faster to create a single array and perform filtering inside a for loop.